src/compiler/checker.ts

```diff
@@ -15014,6 +15014,15 @@ namespace ts {
                     assumeTrue = !assumeTrue;
                 }
                 const valueType = getTypeOfExpression(value);
+                if ((type.flags & TypeFlags.Unknown) && (operator === SyntaxKind.EqualsEqualsEqualsToken) && assumeTrue) {
+                    if (valueType.flags & TypeFlags.Primitive || valueType.flags & TypeFlags.NonPrimitive) {

```

Comment:
Simplify to `valueType.flags & (TypeFlags.Primitive | TypeFlags.NonPrimitive)`

Reply:
Oops, thanks @RyanCavanaugh ! Fixed now